runtime/trappist/src/constants.rs

```diff
@@ -58,8 +60,38 @@ pub mod fee {
 	/// Yet, it can be used for any other sort of change to weight-fee. Some examples being:
 	///   - Setting it to `0` will essentially disable the weight fee.
 	///   - Setting it to `1` will cause the literal `#[weight = x]` values to be charged.
+	///
+	/// TODO: Once the runtime is upgraded to polkadot v0.9.42 or above refactor this using
+	/// the FeePolynomial struct that already includes the methods to make this calculations
+	/// and remove the custom WeightCoefficientCalc inside ./trappist/src/impls.rs
 	pub struct WeightToFee;
-	impl WeightToFeePolynomial for WeightToFee {
+	impl frame_support::weights::WeightToFee for WeightToFee {
+		type Balance = Balance;
+
+		fn weight_to_fee(weight: &Weight) -> Self::Balance {
+			let ref_poly: smallvec::SmallVec<[WeightToFeeCoefficient<Balance>; 4]> =
```

Comment:
Rust can infer the type properly here, so you don't need to specify any. Still if you want you can use `WeightToFeeCoefficients<Self::Balance>` I don't see why it should be known that this is a type alias for smallvec.